File: android/app/src/main/java/com/cochat/app/data/repository/NotificationsRepository.kt
package com.cochat.app.data.repository

import com.cochat.app.data.model.AppNotification
import com.cochat.app.data.remote.ApiService

class NotificationsRepository(private val api: ApiService) {
    suspend fun list(): List<AppNotification> = api.listNotifications().notifications
    suspend fun markRead(id: String) { api.markNotificationRead(id) }
    suspend fun markAllRead() { api.markAllNotificationsRead() }
}

File: ios/CoChat/Repositories/NotificationsRepository.swift
import Foundation

final class NotificationsRepository {
    private let api: APIClient

    init(api: APIClient) {
        self.api = api
    }

    func list() async throws -> [AppNotification] {
        let res: NotificationsEnvelope = try await api.get("notifications")
        return res.notifications
    }

    func markRead(id: String) async throws {
        try await api.postVoid("notifications/\(id)/read")
    }

    func markAllRead() async throws {
        try await api.postVoid("notifications/read-all")
    }
}
